docs/proposals/zeroconf.md

```diff
@@ -0,0 +1,142 @@
+# [ZeroConf](https://en.wikipedia.org/wiki/Zero-configuration_networking) Protocol Implementation
+
+## Goal
+
+Agent implements [Zero-configuration networking](https://en.wikipedia.org/wiki/Zero-configuration_networking) (hence 'ZeroConf'), a set of technologies that help discover devices and services using DNS-based discovery. There are 2 main elements: Multicast DNS (mDNS) and DNS-based Service Discovery (DNS-SD).
+
+While ZeroConf is often used in home networks (that don't often include regular DNS), ZeroConf is broadly applicable and is useful in IoT deployments in which devices are transient, there are many devices, developers wish to dynamically manage services on these devices.
+
+These technologies require additional packages and shared libraries. Supporting ZeroConf as an Akri protcol possibly (!?) provides a mechanism by which (Kubernetes) application developers can leverage ZeroConf technologies without having to install or be familiar with ZeroConf dependencies.
```

Comment:
Is (!?) to convey uncertainty and possibility? Might not be universally understood.

Reply:
Fixed: https://github.com/deislabs/akri/pull/162/commits/91c7004d05fd0ef4de434d5ce0a6480ae8abf44b

And corrected r/protcol/protocol.